net/rpc/server.go

```diff
@@ -462,20 +457,24 @@ func (server *Server) ServeRequest(codec ServerCodec) error {
 		}
 		// send a response if we actually managed to read a header.
 		if req != nil {
-			server.sendResponse(sending, req, invalidRequest, codec, err.Error())
+			server.sendResponse(sending, req, invalidRequest, codec, err)
 			server.freeRequest(req)
 		}
 		return err
 	}
 
-	handler := func() {
-		service.call(server, sending, nil, mtype, req, argv, replyv, codec)
+	handler := func() error {
+		err := service.call(mtype, argv, replyv)
+		server.sendResponse(sending, req, replyv.Interface(), codec, err)
+		server.freeRequest(req)
```

Comment:
hm, could we keep these in `service.call` for the time being?

---

I'd prefer we keep things as structurally the same as much as possible.